src/lib/builders/pin-input/create.ts

```diff
@@ -16,12 +18,14 @@ import type { CreatePinInputProps } from './types';
 
 const { name, selector } = createElHelpers<'input' | 'hidden-input'>('pin-input');
 
-const getInputs = (node: HTMLElement) => {
+const getInputs = (node: HTMLInputElement) => {
 	const rootEl = node.closest(selector());
-	if (!rootEl) return { inputs: null, el: node as HTMLInputElement, elIndex: -1 };
-	const inputs = Array.from(rootEl.querySelectorAll(selector('input'))) as HTMLInputElement[];
+	if (!isHTMLElement(rootEl)) return { inputs: null, el: node, elIndex: -1 };
+	const inputs = Array.from(rootEl.querySelectorAll(selector('input'))).filter(
+		(input): input is HTMLInputElement => isHTMLInputElement(input)
```

Comment:
Do you need the `input is HTMLInputElement` part? That should already be the return type of `isHTMLInputElement` 🤔

Reply:
I know filters are a bit weird and I couldn't get it to type properly without it, but let me check again :)